### ISSUE TITLE
Syntax error in a map with if expression without brackets

### ISSUE THREAD

**Issue by frankweenink** [DeltaXML/vscode-xslt-tokenizer#91]
When using an if expression in a map without brackets it is reported as a problem while it's a valid XPATH expression:

> XPath: Expression context - unexpected token here [line,column]

where [line,column] in this example points to the last colon

```xslt
<xsl:variable name="example" select="map {
      'a' : if (true()) then 1 else 2,
      'b' : 2
  }"/>
```

A workaround is to use brackets around the if expression:
```xslt
  <xsl:variable name="example" select="map {
      'a' : (if (true()) then 1 else 2),
      'b' : 2
  }"/>
````

**Comment by pgfearo** [comment]
Thanks for reporting this issue, it appears related to how the ifElse expression is handled which I'm already seeking to improve. I'll report back when I have more.

**Comment by pgfearo** [comment]
This issue has been fixed in Release 0.3.16
